### ISSUE TITLE
Unclear Size limits

### ISSUE THREAD

**Issue by sasha-wing** [cloudevents/spec#1259]
In the [size limits section](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#size-limits) it is stated 

> * Intermediaries MUST forward events of a size of 64 KByte or less.
> * Consumers SHOULD accept events of a size of at least 64 KByte.
> In effect, these rules will allow producers to publish events up to 64KB

It is not clear if this means 1000 bytes or 1024 bytes

because when i searched "KByte" in google I was presented with the definition of 1000 bytes

![image](https://github.com/cloudevents/spec/assets/113886702/d8907960-4fe7-478e-8eb7-4ad08d11e408)

The wikipedia table also makes it confusing to understand what is the exact size of the limit
![image](https://github.com/cloudevents/spec/assets/113886702/95daaf57-9b2a-4bf8-b168-7a652f6ccd40)

Additionally the size is not consistent, first it is `64 KByte` and later it is `64KB`.

In summary, I think there is a need for clarification here.

**Comment by duglin** [comment]
Per today's call, it should be 1024.  Wanna PR it?

**Comment by sasha-tkachev** [comment]
Yes
